test/stop_when_test.cpp

```diff
@@ -127,3 +135,17 @@ TEST(StopWhen, Pipeable) {
   EXPECT_FALSE(sourceExecuted);
   EXPECT_TRUE(triggerExecuted);
 }
+
+#if !UNIFEX_NO_COROUTINES
+TEST(StopWhen, SynchronousCompletionFromReceiverStopRequestIsASANSafe) {
```

Comment:
Would it repro with `allocate()` too?

Reply:
Mysteriously, no.  It took me a long time to narrow this one down, and I can't get it any smaller.